geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/DestroyConnectionCommand.java

```diff
@@ -54,24 +54,18 @@ public Result destroyConnection(@CliOption(key = DESTROY_CONNECTION__NAME, manda
     // action
     List<CliFunctionResult> results =
         executeAndGetFunctionResult(new DestroyConnectionFunction(), name, targetMembers);
-
-    boolean persisted = false;
-    ClusterConfigurationService ccService = getConfigurationService();
-
-    if (ccService != null && results.stream().filter(CliFunctionResult::isSuccessful).count() > 0) {
-      ConnectorService service =
-          ccService.getCustomCacheElement("cluster", "connector-service", ConnectorService.class);
-      if (service != null) {
-        ConnectorService.Connection conn = CacheElement.findElement(service.getConnection(), name);
-        service.getConnection().remove(conn);
-        ccService.saveCustomCacheElement("cluster", service);
-        persisted = true;
-      }
-    }
-
     CommandResult commandResult = ResultBuilder.buildResult(results);
-    commandResult.setCommandPersisted(persisted);
-
+    CacheElement element = (CacheElement) () -> name;
+    commandResult.setCacheElemnt(element);
     return commandResult;
   }
+
+  @Override
+  public void updateClusterConfig(String group, CacheConfig config, CacheElement element) {
+    ConnectorService service =
+        config.findCustomCacheElement("connector-service", ConnectorService.class);
+    if (service != null) {
```

Comment:
I think if we are attempting to remove a `service` that does not exist in the cluster, this should hard fail rather than fail quietly.

Reply:
the remove after it also won't remove anything if it couldn't find the element with the id. The end goal is that the config element won't exist in the cluster config. Failing to find it in the beginning would be considered a "failed" operation.